host-bin/mount-chroot

```diff
@@ -185,29 +199,27 @@ for NAME in "$@"; do
             echo 'done' 1>&2
 
             # Create key file
-            wrappedkey="`mktemp`"
-            wrappedfnek="`mktemp`"
-            addtrap "rm -f '$wrappedkey' '$wrappedfnek'"
-            echo -n "$key
+            if [ -z "$NOPASS" ]; then
+                wrappedkey="`mktemp`"
+                wrappedfnek="`mktemp`"
+                addtrap "rm -f '$wrappedkey' '$wrappedfnek'"
+                echo -n "$key
 $passphrase" | ecryptfs-wrap-passphrase "$wrappedkey" -
-            echo -n "$fnek
+                echo -n "$fnek
 $passphrase" | ecryptfs-wrap-passphrase "$wrappedfnek" -
-            unset key fnek
-            echo | cat - "$wrappedkey" "$wrappedfnek" > "$KEYFILE"
+                unset key fnek
+                echo | cat - "$wrappedkey" "$wrappedfnek" > "$KEYFILE"
+            else
+                echo -n "
+................$key................$fnek" > "$KEYFILE"
```

Comment:
Can you put the 16 periods into a variable at the top?  Counting them all each time is error-prone.